### PR TITLE
Closes #27 Introduce template code to add centiloid threshold criteria flag

### LINKED ISSUE #27: ADAPET template - implement criterion flags for amyloid categories
@cgpele please could you fill in what you expect to see here? Thanks

### PR DESCRIPTION
**Pull request by cgpele**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: inst/templates/ad_adapet.R [hunk 1]
```diff
@@ -161,6 +161,15 @@ adapet <- adapet %>%
     )
   )
 
+## Apply the derive_vars_crit_flag function for Centiloid Threshold
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:164]:
```suggestion
## Create criterion flags for Centiloid Threshold
```
(no need to explicitly state that you are using `derive_vars_crit_flag()` as it's clear from the code)

File: inst/templates/ad_adapet.R [hunk 2]
```diff
@@ -161,6 +161,15 @@ adapet <- adapet %>%
     )
   )
```
Comment (manciniedoardo) [inline inst/templates/ad_adapet.R:163]: I am wondering whether this new block that you've added could actually be piped directly from the `slice_derivation()` block as they're both amyloid-specific derivations? Then you could organise the block under a section header called "amyloid/neuro-specific derivations". I.e. something like:
```
## Amyloid-specific derivations ----
### Convert SUVR to Centiloid ----
adapet <- adapet %>%
  slice_derivation(
    derivation = derive_param_computed,
    args = params(
      by_vars = c(
        get_admiral_option("subject_keys"),
        adsl_vars,
        exprs(ADT, ADY, VISIT)
      ),
      keep_nas = TRUE
    ),
    derivation_slice(
      filter = (PARAMCD == "SUVRBFBB") & (NVMETHOD == "BERKELEY FBB SUVR PIPELINE") & (REFREG == "Whole Cerebellum"),
      args = params(
        parameters = c("SUVRBFBB"),
        set_values_to = exprs(
          AVAL = compute_centiloid(
            tracer = "18F-Florbetaben",
            pipeline = "BERKELEY FBB SUVR PIPELINE",
            ref_region = "Whole Cerebellum",
            suvr = AVAL
          ),
          PARAMCD = "CLBFBB",
          PARAM = "Centiloid (CL) based on BERKELEY FBB",
          AVALU = "CL"
        )
      )
    ),
    # Can use the following code if data contain "BERKELEY FBP SUVR PIPELINE"
    # Commented out as the the example nv data do not have such pipeline
    # derivation_slice(
    #   filter = (PARAMCD == "SUVRBFBP") & (NVMETHOD == "BERKELEY FBP SUVR PIPELINE") & (REFREG == "Whole Cerebellum"),
    #   args = params(
    #     parameters = c("SUVRBFBP"),
    #     set_values_to = exprs(
    #       AVAL = compute_centiloid(
    #         tracer = "18F-Florbetapir",
    #         pipeline = "BERKELEY FBP SUVR PIPELINE",
    #         ref_region = "Whole Cerebellum",
    #         suvr = AVAL
    #       ),
    #       PARAMCD = "CLBFBP",
    #       PARAM = "Centiloid (CL) based on BERKELEY FBP",
    #       AVALU = "CL"
    #     )
    #   )
    # ),
    derivation_slice(
      filter = (PARAMCD == "SUVRAFBP") & (NVMETHOD == "AVID FBP SUVR PIPELINE") & (REFREG == "Whole Cerebellum"),
      args = params(
        parameters = c("SUVRAFBP"),
        set_values_to = exprs(
          AVAL = compute_centiloid(
            tracer = "18F-Florbetapir",
            pipeline = "AVID FBP SUVR PIPELINE",
            ref_region = "Whole Cerebellum",
            suvr = AVAL
          ),
          PARAMCD = "CLAFBP",
          PARAM = "Centiloid (CL) based on AVID FBP",
          AVALU = "CL"
        )
      )
    )
  ) %>%
### Derive criterion flags for Centiloid Threshold ----
  derive_vars_crit_flag(
    crit_nr = 1,
    condition = if_else(PARAMCD == "CLAFBP", AVAL < 24.1, NA),
    description = "CENTILOID < 24.1 (for CLAFBP)",
    values_yn = TRUE # To get "Y", "N", and NA for the flag
  )
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
@cgpele looking great, thanks! Just left a couple of comments. Also please be sure to save the updated dataset as `admiralneuro_adapet.rda` in the `data` folder.

**Comment by manciniedoardo** [comment]
Added @jwang-lilly for review and removed @meilinjiang who is OOO. Also @pharmaverse/admiralneuro please feel free to take a look! Looking to merge by end of week.

**Comment by jwang-lilly** [comment]
@cgpele, great job. I have no further comments.

File: inst/templates/ad_adapet.R (hunk 2 above)
Comment (cgpele) [reply inst/templates/ad_adapet.R:163]: Happy to implement it that way

**Review by manciniedoardo — APPROVED** [review]
Thanks @cgpele - looks great!

**Comment by manciniedoardo** [comment]
@jwang-lilly wjhen you have a moment, please could you check you're happy and merge? as discussed on wednesday we have an (expected) conflict with admiralneuro_adapet due to Miles' PR being merged, so you'll have to go into branch 27_amyloid_criterion_flag, merge main into that branch, rerun the template and re-save admiralneuro_adapet to solve the conflict.

File: inst/templates/ad_adapet.R [hunk 3]
```diff
@@ -159,8 +160,16 @@ adapet <- adapet %>%
         )
       )
     )
+  ) %>%
+  ### Derive criterion flags for Centiloid Threshold ----
+  derive_vars_crit_flag(
+    crit_nr = 1,
+    condition = if_else(PARAMCD %in% c("CLAFBP", "CLBFBB"), AVAL < 24.1, NA),
+    description = "CENTILOID < 24.1",
```
Comment (Fanny-Gautier) [inline inst/templates/ad_adapet.R:168]: Is it ok to have CRIT1 populated even for the SUVR parameters? 
image

Comment (Lina2689) [reply inst/templates/ad_adapet.R:168]: I have cross checked with ADaM IG v1.3, it seems incorrect to populate CRIT1 if CRIT1FL is **NA**. Here, its confusing because the CRIT1/CRTIT1FL are only to define where the "CLAFBP", "CLBFBB" parameters are lower than 24.1. Its not regarding the SUVR* parameters. So, we need to keep CRIT1/CRIT1FL as blank for all other parameters than the CENTILOID ones.

**Comment by jwang-lilly** [comment]
@cgpele, I just saw the comment from @Lina2689 in response to @Fanny-Gautier, if you are to modify your flag codes, please note:
I merged main into 27_amyloid_criterion_flag, resolved a conflict, and pushed. Now remote branch 27_ has moved ahead of your local branch, the remote branch includes merged PRs 18 and 35, a friendly reminder to do: 
git pull origin 27_amyloid_criterion_flag
prior to modify your flag codes.

**Comment by Lina2689** [comment]
@jwang-lilly CRIT1/CRIT1FL looks good now. 

For NVCAT='FBB' and NVTEST='Qualitative Visual Classification' there are blank PARAM/PARAMCD. Is there any specific reason?
 
image

**Comment by jwang-lilly** [comment]
> @jwang-lilly CRIT1/CRIT1FL looks good now.
> 
> For NVCAT='FBB' and NVTEST='Qualitative Visual Classification' there are blank PARAM/PARAMCD. Is there any specific reason?
Thanks @Lina2689 for catching this.  Lookup table missed FBB visual read.
@cgpele, the lookup table should be changed to the following, please note I added 2nd to the last row: VR, FBB etc.

# Assign PARAMCD, PARAM, and PARAMN
`
param_lookup <- tibble::tribble(
  ~NVTESTCD, ~NVCAT, ~NVLOC, ~REFREG, ~NVMETHOD, ~PARAMCD, ~PARAM, ~PARAMN,
  "SUVR", "FBP", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "AVID FBP SUVR PIPELINE", "SUVRAFBP", "FBP Standard Uptake Ratio Neocortical Composite Whole Cerebellum", 1,
  "SUVR", "FBB", "NEOCORTICAL COMPOSITE", "Whole Cerebellum", "BERKELEY FBB SUVR PIPELINE", "SUVRBFBB", "FBB Standard Uptake Ratio Neocortical Composite Whole Cerebellum", 2,
  "SUVR", "FTP", "NEOCORTICAL COMPOSITE", "Inferior Cerebellar Gray Matter", "BERKELEY FTP SUVR PIPELINE", "SUVRBFTP", "FTP Standard Uptake Ratio Neocortical Composite Inferior Cerebellar Gray Matter", 3,
  "VR", "FBP", NA, NA, "FBP VISUAL CLASSIFICATION", "VRFBP", "FBP Qualitative Visual Classification", 4,
  "VR", "FBB", NA, NA, "FBB VISUAL CLASSIFICATION", "VRFBB", "FBB Qualitative Visual Classification", 5,
  "VR", "FTP", NA, NA, "FTP VISUAL CLASSIFICATION", "VRFTP", "FTP Qualitative Visual Classification", 6
)

`

**Comment by Fanny-Gautier** [comment]
To be confirmed which `lookup_param` table is to be used. Please take a look at my [comment](https://github.com/pharmaverse/admiralneuro/issues/42#issuecomment-3112567718) in issue 42
